docs/rfc/active/5818-serial-updates.md

```diff
@@ -0,0 +1,313 @@
+* A client could include multiple aspects in its precondition state, in case one update relies on the state of many 
+  other aspects. We need to defend against race conditions for all the aspects given.
+
+### Extensibility
+
+> Please also call out extensibility requirements. Is this proposal meant to be extended in the future? Are you adding
+> a new API or set of models that others can build on in later? Please list these concerns here as well.
+
+1. The proposal could be extended to include a list of aspect versions which must hold true in order for a single 
+   aspect update to occur.
+2. We could extend to include the state version in a batch update, though this may be complicated if handling 
+   multiple updates to the same aspect in a single batch.
+
+## Non-Requirements
+
+It's not important for us to discuss complex prerequisites here. The only thing we need to enforce is the state of 
+an aspect has not changed when making an update to it.
+
+There are some potential spin-offs of this design which could involve writing some kind of PATCH update where a client 
+supplies a diff instead of a complete new state, but this is out of scope of this particular RFC.
+
+## Detailed design
+
+> This is the bulk of the RFC.
+
+> Explain the design in enough detail for somebody familiar with the framework to understand, and for somebody familiar
+> with the implementation to implement. This should get into specifics and corner-cases, and include examples of how the
+> feature is used. Any new terminology should be defined here.
+
+In short, my recommendations for implementing this change is proposal 2 or 3 in the list below. But first, a 
+consideration on versions...
+
+#### Aspect Versions
+
+There are many ways to achieve this, but most of them involve DataHub aspects having some kind of consistent 
+versioning as currently the LATEST version of an aspect is version zero and is a mutable placeholder for a "real" 
+fixed version.
+
+We don't necessarily have to overhaul the versioning approach here, as all we need is a way of uniquely identifying 
+the aspect instance that is to be changed. We don't need to rely on numerical version ordering for this, but 
+something will need to be added as the general aspect level to allow for this new functionality.
+
+Whatever is done, the most important thing is that changes can be isolated at the metadata storage level, whether 
+through locking, transactions or Cassandra LWTs. More on this later.
+
+### 1: Change Processor
+
+Check the allowed set of transitions between current and proposed aspect state. This would be achieved with 
+business-specific logic injected into the DataHub codebase via plugins. This feels like teh wrong approach.
+
+Good
+* Fast as all code is executed in GMS
+
+Bad
+* No guarantee of state between the read and write
+* No guarantee that the plugin code will run
+* Specific business logic getting tied up with GMS code
```

Comment:
This is the biggest issue with this. I think most companies using DataHub simply wouldn't bother with this, and thus would not benefit from it!